bus-mapping/src/evm/opcodes/stackonlyop.rs

```diff
@@ -145,28 +168,237 @@ mod stackonlyop_tests {
+                PUSH1(b)
+                PUSH1(a)
+                OR
+                STOP
+            },
+            vec![
+                StackOp::new(1, StackAddress(1022), Word::from(a)),
+                StackOp::new(1, StackAddress(1023), Word::from(b)),
+            ],
+            vec![StackOp::new(1, StackAddress(1023), Word::from(result))],
+        );
+    }
     #[test]
-    fn addmod_opcode_impl() {
-        stack_only_opcode_impl::<3, 1>(
-            OpcodeId::ADDMOD,
+    fn test_or_operate() {
+        or_opcode_impl(0x01, 0x01, 0x01.bitor(0x01) as usize);
+        or_opcode_impl(0x00, 0x01, 0x00.bitor(0x01) as usize);
+        or_opcode_impl(0x00, 0x00, 0x00.bitor(0x00) as usize);
+    }
+
+    fn xor_opcode_impl(a: usize, b: usize, result: usize) {
+        stack_only_opcode_impl::<2, 1>(
+            OpcodeId::XOR,
             bytecode! {
-                PUSH3(0xbcdef)
-                PUSH3(0x6789a)
-                PUSH3(0x12345)
-                ADDMOD
+                PUSH1(b)
+                PUSH1(a)
+                XOR
                 STOP
             },
             vec![
-                StackOp::new(1, StackAddress(1021), Word::from(0x12345)),
-                StackOp::new(1, StackAddress(1022), Word::from(0x6789a)),
-                StackOp::new(1, StackAddress(1023), Word::from(0xbcdef)),
+                StackOp::new(1, StackAddress(1022), Word::from(a)),
+                StackOp::new(1, StackAddress(1023), Word::from(b)),
             ],
-            vec![StackOp::new(1, StackAddress(1023), Word::from(0x79bdf))],
+            vec![StackOp::new(1, StackAddress(1023), Word::from(result))],
+        );
+    }
+    #[test]
+    fn test_xor_operate() {
+        xor_opcode_impl(0x01, 0x01, 0x01.bitxor(0x01) as usize);
+        xor_opcode_impl(0x01, 0x00, 0x00.bitxor(0x01) as usize);
+        xor_opcode_impl(0x00, 0x00, 0x00.bitxor(0x00) as usize);
+    }
+
+    #[test]
+    fn not_opcode_impl() {
+        stack_only_opcode_impl::<1, 1>(
+            OpcodeId::NOT,
+            bytecode! {
+                PUSH32(word!("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"))
+                NOT
+                STOP
+            },
+            vec![StackOp::new(
+                1,
+                StackAddress(1023),
+                word!("0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"),
+            )],
+            vec![StackOp::new(
+                1,
+                StackAddress(1023),
+                word!("0xfffefdfcfbfaf9f8f7f6f5f4f3f2f1f0efeeedecebeae9e8e7e6e5e4e3e2e1e0"),
+            )],
+        );
+    }
+
+    // 0x44 DIFFICULTY
+    #[test]
+    fn difficulty_opcode_impl() {
+        stack_only_opcode_impl::<0, 1>(
+            OpcodeId::DIFFICULTY,
+            bytecode! {
+                DIFFICULTY
+                STOP
+            },
+            vec![],
+            vec![StackOp::new(1, StackAddress(1023), word!("0x200000"))],
```

Comment:
Can you get the difficulty param from the testing constants instead of hardcoding it??

Reply:
This is a good idea. Thanks.